### ISSUE TITLE
W605: False positive on line continuation in triple-quote string

### ISSUE THREAD

**Issue by thejcannon** [PyCQA/pycodestyle#974]
Given this code (from the Python docs: https://docs.python.org/3/library/textwrap.html):

```python
def test():
    # end first line with \ to avoid the empty line!
    s = '''\
    hello
      world
    '''
    print(repr(s))          # prints '    hello\n      world\n    '
    print(repr(dedent(s)))  # prints 'hello\n  world\n'
```

I'm seeing W605 on the line continuation (which looks suspiciously like an escape sequence, but isn't).

**Comment by asottile** [comment]
I cannot reproduce:

```console
$ ~/opt/venv/bin/pycodestyle -
def test():
    # end first line with \ to avoid the empty line!
    s = '''\
    hello
      world
    '''
    print(repr(s))          # prints '    hello\n      world\n    '
    print(repr(dedent(s)))  # prints 'hello\n  world\n'
$ echo $?
0
```

**Comment by asottile** [comment]
make sure you're on the latest version of pycodestyle and that you don't have trailing whitespace after your backslash (which _would_ be an invalid escape sequence)

**Comment by irwand** [comment]
Looks like this is only a problem when flake8 is called to process a file through PIPE. Example:
```
C:\dev\Tools\eng\pipeline\nibuild> \dev\.vimvenv\Scripts\flake8.exe --format=def
ault --stdin-display-name "_pipeline" _pipeline.py

C:\dev\Tools\eng\pipeline\nibuild> \dev\.vimvenv\Scripts\flake8.exe --format=def
ault --stdin-display-name "_pipeline" - < _pipeline.py
'pipeline:850:13: W605 invalid escape sequence '\
        f"""\
        workspace {name} ({name}) {version} Export
        RUNTIME DEPENDENCIES
        Running this export requires the following components (or newer
        versions not breaking backwards compatibility) to be installed:
        None.
        BUILD DEPENDENCIES
        The trunk is built against the following exports:
        None
        """
            ^
```

The piping through file is what ALE does when linting the file through vim, which is how this problem reproduce for me. :'(

**Comment by asottile** [comment]
please show your version information and the exact bytes of the file (you can attach a zip) -- I cannot reproduce, even on windows

**Comment by irwand** [comment]
I also can't reproduce it calling pycodestyle directly.. only shows up when calling flake8 with PIPE:

```
(.vimvenv) C:\dev\Tools\eng\pipeline\nibuild> pycodestyle.exe - < \tmp\w605.py

(.vimvenv) C:\dev\Tools\eng\pipeline\nibuild> flake8 - < \tmp\w605.py
'tdin:3:20: W605 invalid escape sequence '\
s = text.dedent("""\
    hello
      world
    """)
                   ^

(.vimvenv) C:\dev\Tools\eng\pipeline\nibuild> cat \tmp\w605.py
import text

s = text.dedent("""\
    hello
      world
    """)
print(s)

(.vimvenv) C:\dev\Tools\eng\pipeline\nibuild> pip freeze
appdirs==1.4.4
black==20.8b1
click==7.1.2
dataclasses==0.8
flake8==3.8.4
importlib-metadata==3.3.0
mccabe==0.6.1
mypy-extensions==0.4.3
pathspec==0.8.1
pycodestyle==2.6.0
pyflakes==2.2.0
regex==2020.11.13
toml==0.10.2
typed-ast==1.4.1
typing-extensions==3.7.4.3
zipp==3.4.0

(.vimvenv) C:\dev\Tools\eng\pipeline\nibuild> python --version
Python 3.6.8
```

**Comment by irwand** [comment]
@asottile : Thank you for being VERY responsive btw.. 👍 :-)

**Comment by asottile** [comment]
can you attach a zip of that file?  it will help understand what's going on

**Comment by irwand** [comment]
Here's the zip file: 
[w605.zip](https://github.com/PyCQA/pycodestyle/files/5782556/w605.zip)

It contains:
* w605.py
* reproduce.bat <- just run this maybe?
* .vimvenv/  <- my vimvenv that I used. Not sure how well this would work on your machine tho

NOTE: 
* unzip under C:\dev location in Windows
* if .vimvenv doesn't work, feel free to rebuild it. I used python 3.6.8, install flake8

**Comment by asottile** [comment]
hmmm interesting:

```console
$ hd -c w605.py 
00000000  69 6d 70 6f 72 74 20 74  65 78 74 0d 0a 0d 0a 73  |import text....s|
0000000   i   m   p   o   r   t       t   e   x   t  \r  \n  \r  \n   s
00000010  20 3d 20 74 65 78 74 2e  64 65 64 65 6e 74 28 22  | = text.dedent("|
0000010       =       t   e   x   t   .   d   e   d   e   n   t   (   "
00000020  22 22 5c 0d 0a 20 20 20  20 68 65 6c 6c 6f 0d 0a  |""\..    hello..|
0000020   "   "   \  \r  \n                   h   e   l   l   o  \r  \n
00000030  20 20 20 20 20 20 77 6f  72 6c 64 0d 0a 20 20 20  |      world..   |
0000030                           w   o   r   l   d  \r  \n            
00000040  20 22 22 22 29 0d 0a 70  72 69 6e 74 28 73 29 0d  | """)..print(s).|
0000040       "   "   "   )  \r  \n   p   r   i   n   t   (   s   )  \r
00000050  0a                                                |.|
0000050  \n                                                            
0000051
```

crlf newlines seem to be throwing this off, but only on windows?


interestingly I've found another, related bug due to `tokenize.detect_encoding` -- on the first few lines of the file pycodestyle sees the `\r` but not on the rest:

```python
'''\
'''
```

```console
$ unix2dos t.py
$ pycodestyle t.py
'.py:1:4: W605 invalid escape sequence '\
```

the other one I think is a bug in how flake8 treats stdin vs files, I can fix that from the flake8 side

**Comment by asottile** [comment]
#976 for the new bug and https://gitlab.com/pycqa/flake8/-/merge_requests/461 for the flake8 bug

**Comment by irwand** [comment]
oh, interesting.. yes, I concur with your findings.
I tried this on both Windows and Linux. Looks like crlf confuses flake8 when using PIPE, only on Windows..
If I change the w605.py to use unix line ending, the problem doesn't come up.

**Comment by asottile** [comment]
cool, merged the patch for the beginning-of-file issue and I'll finish this up in flake8 -- thanks for the additional details! (just a tip for next time, including all of that information up front is extremely helpful so that we can take your problem, reproduce it, and fix it -- it saves everyone time and frustration)

**Comment by irwand** [comment]
Thanks man for the fix! You rock!! Thanks for the tips.. 👍 Have a great one!
